ktor-client/ktor-client-features/ktor-client-auth-basic/src/io/ktor/client/features/auth/basic/BasicAuth.kt

```diff
@@ -0,0 +1,47 @@
+package io.ktor.client.features.auth.basic
+
+import io.ktor.client.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.util.*
+
+class BasicAuth(val username: String, val password: String) {
+
+  class Configuration {
+
+    var username: String = ""
+    var password: String = ""
```

Comment:
I'd prefer to keep configuration uninitialized and use `lateinit var` instead(It helps to check missing initializers at configuration phase)